src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,659 @@
+    dsError_t status   = dsERR_NONE;
+    int32_t frfMode = 0;
+
+    dsVideoDevice_getHandle();
+    UT_LOG_INFO("Calling dsGetFRFMode(IN:Handle[%d],OUT:frfMode)",gHandle);
+    status = dsGetFRFMode(gHandle, &frfMode);
+    UT_LOG_INFO("Result dsGetFRFMode(), OUT:dsError_t=[%s] OUT:frfMode=[0x08%d]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), frfMode);
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Get Current Frame Rate 
+ *
+ * This test gets Current Frame Rate of the device
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 009@n
+ * **Pre-Conditions:** None@n
+ * VideoDevice_init Should be called before calling this test
+ *
+ * **Dependencies:** None@n
+ *
+ * **User Interaction:** @n
+ * User or Automation tool should selected GetCurrentDisplayframerate from list
+ *
+ */
+void test_l3_dsVideoDevice_dsGetCurrentDisplayframerate()
+{
+    gTestID = 9;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t status   = dsERR_NONE;
+    char currentFrameRate[dsVIDEO_FRAMERATE_MAX];
+
+    dsVideoDevice_getHandle();
+    UT_LOG_INFO("Calling dsGetCurrentDisplayframerate(IN:Handle[%d],OUT:currentFrameRate)",gHandle);
+    status = dsGetCurrentDisplayframerate(gHandle, currentFrameRate);
+    UT_LOG_INFO("Result dsGetCurrentDisplayframerate(), OUT:dsError_t=[%s] OUT:currentFrameRate=[%s]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), currentFrameRate);
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+ * @brief Get Current Zoom Mode 
+ *
+ * This test gets Current Zoom Mode of the device
+ * **Test Group ID:** 03@n
+ * **Test Case ID:** 010@n
+ * **Pre-Conditions:** None@n
+ * VideoDevice_init Should be called before calling this test
+ *
+ * **Dependencies:** None@n
+ *
+ * **User Interaction:** @n
+ * User or Automation tool should selected GetZoomMode from list
+ *
+ */
+void test_l3_dsVideoDevice_dsGetZoomMode()
+{
+    gTestID = 10;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t status   = dsERR_NONE;
+    dsVideoZoom_t dfc;
+
+    dsVideoDevice_getHandle();
+    UT_LOG_INFO("Calling dsGetDFC(IN:Handle[%d],OUT:dfc)",gHandle);
+    status = dsGetDFC(gHandle, &dfc);
+    UT_LOG_INFO("Result dsGetDFC(), OUT:dsError_t=[%s] OUT:CurrentZoomMode=[%s]", \
+                                                UT_Control_GetMapString(dsErrorMappingTable, status), \
+                                                UT_Control_GetMapString(dsVideoZoomMappingTable,(int32_t) dfc));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+
+int test_l3_dsVideoDevice_register(void)
```

Comment:
please do the register in test_register.c

Reply:
Updated.